src/Symfony/Component/ExpressionLanguage/Tests/ParserTest.php

```diff
@@ -161,4 +161,44 @@ private function createGetAttrNode($node, $item, $type)
     {
         return new Node\GetAttrNode($node, new Node\ConstantNode($item), new Node\ArgumentsNode(), $type);
     }
+
+    /**
+     * @expectedException \Symfony\Component\ExpressionLanguage\SyntaxError
+     */
+    public function testParseWithStringAsInvalidName()
+    {
+        $lexer = new Lexer();
+        $parser = new Parser(array());
+        $parser->parse($lexer->tokenize('foo."#"'), array('foo'));
+    }
```

Comment:
Actually, all of these cases are very similar, so I'd suggest we use [a data provider here](https://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers).